File: src/Web/UserInterface/src/service/MessageService.js
export default class MessageService {
    constructor(request) {
        this.request = request;
    }

    async getList(filter = {}) {
        const response = await this.request({
            method: 'get',
            url: '/message/list',
            params: filter
        });
        return response?.data ?? response;
    }

    async create(payload) {
        await this.request({
            method: 'post',
            url: '/message/create',
            data: payload
        });
    }

    async update(payload) {
        await this.request({
            method: 'post',
            url: '/message/update',
            data: payload
        });
    }

    async delete(messageId) {
        return this.request({ method: 'delete', url: `/message/delete/${messageId}` });
    }
}

File: src/Mobile/lib/services/message_service.dart
import 'package:dio/dio.dart';

class MessageService {
  MessageService(this._dio);

  final Dio _dio;

  Future<dynamic> getList([Map<String, dynamic>? filter]) async {
    final r = await _dio.get<dynamic>('/message/list', queryParameters: filter);
    return r.data;
  }
}
